### PR TITLE
Fix z3 oddity: reset vs optimization

### LINKED ISSUE #611: reentrancy_symbolic.py SolverException
### OS / Environment
ubuntu 16.04

### Manticore version
77b0542b83bce9695aa79d7447aa58f5c056b111

### Python version
2.7

### Actual behavior
```
[+] user deposited some.
[+] Initial world state
     attacker_account 1bfa530d5d685155e98cd7d9dd23f7b6a801cfef balance: 100000000000000000
     exploit_account d51f25e60490392aa9eb72624f93de30ccd111f3 balance: 0
     user_account d30a286ec6737b8b2a6a7b5fbb5d75b895f62956 balance: 0
     contract_account 67518339e369ab3d591d3569ab0a0d83b2ff5198 balance: 100000000000000000
[+] Setup the exploit
         Setting 30 reply reps
         Setting reply string
[+] Attacker first transaction
[+] Attacker second transaction
[+] The attacker destroys the exploit contract and profit
[+] There are 87 alive states now
Traceback (most recent call last):
  File "reentrancy_symbolic.py", line 122, in <module>
    seth.report(state_id)
  File "/mnt/hgfs/code/manticore/manticore/seth.py", line 601, in report
    m, M = solver.minmax(world.constraints, arithmetic_simplifier(account['balance']))
  File "/mnt/hgfs/code/manticore/manticore/core/smtlib/solver.py", line 102, in minmax
    m = self.min(constraints, x, iters)
  File "/mnt/hgfs/code/manticore/manticore/core/smtlib/solver.py", line 97, in min
    return self.optimize(constraints, X, 'minimize')
  File "/mnt/hgfs/code/manticore/manticore/core/smtlib/solver.py", line 433, in optimize
    while self._check() == 'sat' :
  File "/mnt/hgfs/code/manticore/manticore/core/smtlib/solver.py", line 265, in _check
    raise SolverException(_status)
manticore.core.smtlib.solver.SolverException: (objectives
 (V_12  (interval 99999999781240825 100000000000440738))
)
```

### PR DESCRIPTION
**Pull request by feliam**
Fixes #611 

smtlib `(reset)` command wont delete the goals.
Any previous `minimize` or `maximize` will be still in operation after the `(reset)`. When a variable with the same name comes to life again z3 will print `(objectives ...)`
The default printing is changed in z3 Version 4.5.x.
 
Mitigated by just disabling `reset`

### CODE REVIEW

File: manticore/core/smtlib/solver.py [hunk 1]
```diff
@@ -126,9 +126,14 @@ def __init__(self):
         self.support_reset = True
         logger.debug('Z3 version: %s', self.version)
 
-        if self.version >= Version(4, 4, 1):
+        if self.version >= Version(4, 5, 0):
+            self.support_maximize = False
+            self.support_minimize = False
```
Comment (yan) [inline manticore/core/smtlib/solver.py]: was `maximize`/`minimize` support removed in 4.5.0?

File: manticore/core/smtlib/solver.py [hunk 2]
```diff
@@ -492,7 +497,8 @@ def get_value(self, constraints, expression):
 
         self._send('(get-value (%s))'%var.name)
         ret = self._recv()
-        assert ret.startswith('((') and ret.endswith('))')
+        if  not ( ret.startswith('((') and ret.endswith('))') ):
+            raise Exception('SMTLIB error parsing response: %s' % ret)
```
Comment (yan) [inline manticore/core/smtlib/solver.py]: `SolverException`?

File: manticore/core/smtlib/solver.py (hunk 1 above)
Comment (feliam) [reply manticore/core/smtlib/solver.py]: No, they just changed the format. Stuff that was printed out by default is not  any more. And now there is a special separated command to ask for "objectives"

Comment (feliam) [reply manticore/core/smtlib/solver.py]: Should we add an issue for adding support for the new z3 ? 
(Our current parsing wont work)

**Pull request merged**